rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSourceCreationWindow.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.smartgwt.client.widgets.Label;
+import com.smartgwt.client.widgets.form.DynamicForm;
+import com.smartgwt.client.widgets.form.fields.CheckboxItem;
+import com.smartgwt.client.widgets.form.fields.FormItem;
+import com.smartgwt.client.widgets.form.fields.SpacerItem;
+import com.smartgwt.client.widgets.form.fields.TextItem;
+import com.smartgwt.client.widgets.form.fields.UploadItem;
+
+
+/**
+ * Dialog window to create a node source. Downloads dynamically supported
+ * infrastructures and policies when the window is created.
+ */
+public class NodeSourceCreationWindow extends NodeSourceWindow {
+
+    public NodeSourceCreationWindow(RMController controller) {
+        super(controller, "Add Node Source", "Updating available Infrastructures and Policies");
+        buildForm();
+    }
+
+    @Override
+    protected boolean isNodeSourceEdited() {
+        return false;
+    }
+
+    @Override
+    protected void populateFormValues(Label windowLabel, DynamicForm windowForm, TextItem nodeSourceNameItem,
+            CheckboxItem nodesRecoverableItem) {
+
+        nodesRecoverableItem.setValue(true);
+
+        this.controller.fetchSupportedInfrastructuresAndPolicies(() -> {
+
+            LinkedHashMap<String, String> selectItemValues = new LinkedHashMap<>();
+
+            ArrayList<FormItem> allFormItems = prepareFormItems();
+
+            allFormItems.add(this.infrastructureSelectItem);
+            addAllPluginValuesToAllFormItems(allFormItems,
+                                             selectItemValues,
+                                             this.controller.getModel().getSupportedInfrastructures().values());
+            this.infrastructureSelectItem.setValueMap(selectItemValues);
+
+            allFormItems.add(new SpacerItem());
+            selectItemValues.clear();
+
+            allFormItems.add(this.policySelectItem);
+            addAllPluginValuesToAllFormItems(allFormItems,
+                                             selectItemValues,
+                                             this.controller.getModel().getSupportedPolicies().values());
+            this.policySelectItem.setValueMap(selectItemValues);
+
+            this.infrastructureSelectItem.addChangedHandler(changedEvent -> resetFormForInfrastructureSelectChange());
+            this.policySelectItem.addChangedHandler(changedEvent -> resetFormForPolicySelectChange());
+
+            windowForm.setFields(allFormItems.toArray(new FormItem[allFormItems.size()]));
+            windowLabel.hide();
+            windowForm.show();
+
+            hideAllFormItems();
+
+        }, () -> this.window.hide());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)